Xamarin.PropertyEditing/Common/FilePath.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+
+namespace Xamarin.PropertyEditing.Common
+{
+	public class FilePath : IEquatable<FilePath>
+	{
+		public string Source { get; set; }
+
+		public FilePath () { }
+
+		public FilePath (string source) 
```

Comment:
Let's add a `ArgumentNullException` here for `source`, otherwise we might crash later in `GetHashCode()` if it's null.

Reply:
Fixed

Reply:
You fixed this particular request, but it can still NRE in `GetHashCode()` since `Source` is settable and we have an argumentless `ctor`. Since `FilePath` itself is a reference type, I don't think we need to support a null `Source`, so let's just make `FilePath` immutable.